### ISSUE TITLE
block form breaks all other text options? / conditional on promise state in the block?

### ISSUE THREAD

**Issue by nruth** [DockYard/ember-async-button#57]
# Problem
(on master) The following doesn't change to "Saving" when clicked, it just disables the button until the promise resolves

```hbs
{{#async-button action="saveCalculation"
                class='btn btn-primary btn-block btn-lg'
                pending="Saving"
}}
    Save <i class='glyphicon glyphicon-transfer'></i> <i class='glyphicon glyphicon-cloud'></i>
{{/async-button}}
```

I've put it into this jsbin that I copied from another ticket. I couldn't update it to master, but it demonstrates the problem: the button gets disabled but the text does not change.

http://jsbin.com/nigaxinusi/edit?html,js,output

The block form is nice because it allows entering HTML entities, but the readme doesn't say that it breaks the other text options. It says "The component can also take a block".

Has anyone worked with / around this? 

# Solution

1- Update the readme to say block form doesn't support text options for different load states
2- Let people work with the loading state in their block

I think the general solution would be to have a property we can query in the block to decide what to render, as pending might look quite different from default, e.g. having a loading spinner

```hbs
{{#async-button action="saveCalculation"
                class='btn btn-primary btn-block btn-lg'
                pending="Saving"
}}
  {{#if pending}}
    Saving <i class='glyphicon glyphicon-whatever'></i>
  {{#else}}
    Save <i class='glyphicon glyphicon-transfer'></i> <i class='glyphicon glyphicon-cloud'></i>
  {{/if}}
{{/async-button}}
```

Passing "text" into the block as a parameter would help in some cases, as below, but is less general as we wouldn't be able to change the decorating HTML (so no loading spinner).

```hbs
{{#async-button action="saveCalculation"
                class='btn btn-primary btn-block btn-lg'
                pending="Saving"
}}
  {{text}} <i class='glyphicon glyphicon-transfer'></i> <i class='glyphicon glyphicon-cloud'></i>
{{/async-button}}
```

Thanks for reading!

**Comment by drspaniel** [comment]
This is solved with #52 & #71. Will close on merge

**Comment by nruth** [comment]
Great. Thank you for the update!
